### PR TITLE
[netcdf-c/nvtt] Update to the latest version

### PR DESCRIPTION
**Pull request by NancyLi1013**
Update to the latest version

Related #13536

Note: No feature needs to test.

### CODE REVIEW

File: ports/nvtt/squish.patch [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+index 65a5dd3..d093ce5 100644
+--- a/src/nvtt/squish/CMakeLists.txt
 +++ b/src/nvtt/squish/CMakeLists.txt
-@@ -33,3 +33,7 @@ IF(NOT WIN32)
-     ENDIF(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
- ENDIF(NOT WIN32)
+@@ -21,3 +21,8 @@ SET(SQUISH_SRCS
+ 	simd_ve.h)
  
+ ADD_LIBRARY(nvsquish STATIC ${SQUISH_SRCS})
```
Comment (ras0219-msft) [inline ports/nvtt/squish.patch:8]: Does this library also need to be installed?

Comment (NancyLi1013) [reply ports/nvtt/squish.patch:8]: Yes, it seems to be needed by OSG. Please see here https://github.com/microsoft/vcpkg/pull/6611#issue-282096323.

**Comment by NancyLi1013** [comment]
nvtt: x86-windows failed like this:

```
nvcc fatal   : 32 bit compilation is only supported for Microsoft Visual Studio 2013 and earlier
CMake Error at cuda_compile_1_generated_CompressKernel.cu.obj.Debug.cmake:220 (message):
  Error generating
  D:/buildtrees/nvtt/x86-windows-dbg/src/nvtt/CMakeFiles/cuda_compile_1.dir/cuda/./cuda_compile_1_generated_CompressKernel.cu.obj
```
So I will add `Supports: !x86` to skip the check on CI.

**Review by PhoebeHui — APPROVED** [review]
LGTM

**Comment by NancyLi1013** [comment]
`nvtt `failed on osx is due to this error:

```
In file included from /Users/vagrant/Data/buildtrees/nvtt/src/2188752e3f-1c6503c2c5.clean/src/nvtt/icbc.cpp:3:
/Users/vagrant/Data/buildtrees/nvtt/src/2188752e3f-1c6503c2c5.clean/src/nvtt/icbc.h:2797:12: error: always_inline function '_pdep_u32' requires target feature 'bmi2', but would be inlined into function 'interleave' that is compiled without support for 'bmi2'
    return _pdep_u32(a, 0x55555555) | _pdep_u32(b, 0xaaaaaaaa);
           ^
/Users/vagrant/Data/buildtrees/nvtt/src/2188752e3f-1c6503c2c5.clean/src/nvtt/icbc.h:2797:39: error: always_inline function '_pdep_u32' requires target feature 'bmi2', but would be inlined into function 'interleave' that is compiled without support for 'bmi2'
    return _pdep_u32(a, 0x55555555) | _pdep_u32(b, 0xaaaaaaaa);
                                      ^
/Users/vagrant/Data/buildtrees/nvtt/src/2188752e3f-1c6503c2c5.clean/src/nvtt/icbc.h:656:12: error: always_inline function '_mm256_fmadd_ps' requires target feature 'fma', but would be inlined into function 'vmadd' that is compiled without support for 'fma'
    return _mm256_fmadd_ps(a, b, c);
           ^
/Users/vagrant/Data/buildtrees/nvtt/src/2188752e3f-1c6503c2c5.clean/src/nvtt/icbc.h:666:12: error: always_inline function '_mm256_fmsub_ps' requires target feature 'fma', but would be inlined into function 'vmsub' that is compiled without support for 'fma'
    return _mm256_fmsub_ps(a, b, c);
           ^
4 errors generated.
```
But I cannot reproduce this on my local machine.

**Comment by JackBoosY** [comment]
@strega-nil Could you please take a look at osx failure?

Thanks.

**Comment by NancyLi1013** [comment]
@strega-nil 
I try to rebuild this on our local osx machine. But I still cannot reproduce it now. 
I also checked the config and build log and I didn't found any exception about this. 

The osx machine used the same compiler as that in CI.

```
-- The C compiler identification is AppleClang 12.0.0.12000032
-- The CXX compiler identification is AppleClang 12.0.0.12000032
```
I have no idea about this issue now.

Could you please help take a look and give some suggestions?

Thanks.

**Comment by strega-nil** [comment]
@NancyLi1013 weird; I was able to repro it in the VM, though, so there's clearly an issue.

Use the VM that's on vcpkgmm-03; you can get into it by accessing:

```
vcpkgmm-03$ cd ~/vagrant/vcpkg-eg-mac
vcpkgmm-03$ vagrant ssh
```

and then you'll find the vcpkg root with this SHA checked out at ~/vcpkg in the VM.
Make sure to remember to set up your environment variables inside the VM:

```sh
$ export VCPKG_DEFAULT_BINARY_CACHE=/Users/vagrant/Data/archives
$ export VCPKG_DOWNLOADS=/Users/vagrant/Data/downloads
$ export WORKING_ROOT=/Users/vagrant/Data
```

**Comment by NancyLi1013** [comment]
Thanks for your help and suggestions @strega-nil . I will try to reproduce this first according to your above guidelines.

**Comment by NancyLi1013** [comment]
It seems that we have no permission to access the VM machine vcpkgmm-03. I try to connect to this machine and it failed like this:

```
Password:
Access denied
Using keyboard-interactive authentication.
Password:
```

**Comment by PhoebeHui** [comment]
Try to fix the issue on osx by adding compile option '-mbmi2 -mfma' to nvtt on osx.

See more information about the failure: https://stackoverflow.com/questions/59590790/how-to-enable-instrinsic-functions-from-the-preprocessor

**Comment by NancyLi1013** [comment]
Thanks for your help to fix this issue on osx @PhoebeHui.

Sorry for my mistake to use the wrong command.

**Review by strega-nil — APPROVED** [review]

**Comment by BillyONeal** [comment]
@NancyLi1013 I think Robert's code review comment is that you are patching in an `add_library` but aren't patching in an equivalent `install` to install that library. Can you confirm this is doing the right thing for `install`? /cc @ras0219  @ras0219-msft

**Comment by NancyLi1013** [comment]
@BillyONeal 
Thanks for your review and pointing this out.

It seems that the targets name installed should be `nvsquish `instead of `squish `, which is different between version 2.1.1 and 2.1.2.

I will update it soon.

File: ports/nvtt/squish.patch [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
++
 +INSTALL(TARGETS squish
```
Comment (NancyLi1013) [inline ports/nvtt/squish.patch:10]:
```suggestion
+INSTALL(TARGETS nvsquish
```

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**